autotest.pm

```diff
@@ -298,7 +300,10 @@ sub runalltests {
                 make_snapshot($t->{fullname});
             }
 
-            eval { $t->runtest; };
+            eval {
+                $t->runtest;
+                $serial_file_pos = parse_serial_output($serial_file_pos, $t);
```

Comment:
Since you are passing the test reference, i guess this method can actually go to the test object, which would call the backend specific function (if available, or the opposite way around)

Reply:
Done.

validation runs:

1-
Code:

> $testapi::distri->set_expected_serial_failures(soft=>[
>                                                    quotemeta("gcc version 4.8.5 (SUSE Linux)")],
>  );

http://tragicbox.suse.cz/tests/1208#step/welcome/10

2- 
Code: 

> $testapi::distri->set_expected_serial_failures(soft=>[
>                                                    quotemeta("gcc version 4.8.5 (SUSE Linux)")],
>                                                hard=>['ERROR']
>  );

http://tragicbox.suse.cz/tests/1215#step/welcome/14


3- 
Code: 

> $testapi::distri->set_expected_serial_failures(
>                                                hard=>['ERROR']
>  );

http://tragicbox.suse.cz/tests/1216#step/welcome/12